tests/core/v5_1/alexandria/test_alexandria_client.py

```diff
@@ -469,15 +455,122 @@ async def _respond():
 
     with pytest.raises(trio.TooSlowError):
         await alice_alexandria_client.advertise(
             bob.node_id, bob.endpoint, advertisements=advertisements,
         )
+
+
+@pytest.mark.trio
+async def test_alexandria_client_send_locate(bob, bob_network, alice_alexandria_client):
+    async with bob_network.dispatcher.subscribe(TalkRequestMessage) as subscription:
+        await alice_alexandria_client.send_locate(
+            bob.node_id, bob.endpoint, content_key=b"\x01test-key", request_id=b"\x01",
+        )
+        with trio.fail_after(1):
+            talk_response = await subscription.receive()
+        assert talk_response.request_id == b"\x01"
+        message = decode_message(talk_response.message.payload)
+        assert isinstance(message, LocateMessage)
+        assert message.payload.content_key == b"\x01test-key"
+
+
+@pytest.mark.trio
+async def test_alexandria_client_send_locations_single_message(
+    alice, bob, bob_network, alice_alexandria_client
+):
+    advertisements = (AdvertisementFactory(private_key=alice.private_key),)
+
+    async with bob_network.dispatcher.subscribe(TalkResponseMessage) as subscription:
+        await alice_alexandria_client.send_locations(
+            bob.node_id, bob.endpoint, advertisements=advertisements, request_id=b"\x01"
+        )
+        with trio.fail_after(1):
+            talk_request = await subscription.receive()
+        assert talk_request.request_id == b"\x01"
+        message = decode_message(talk_request.message.payload)
+        assert isinstance(message, LocationsMessage)
+        assert message.payload.total == 1
+        assert message.payload.locations == advertisements
+
+
+@pytest.mark.trio
+async def test_alexandria_client_locate_single_response(
+    alice, bob, alice_alexandria_client, bob_alexandria_client
+):
+    advertisements = (AdvertisementFactory(private_key=alice.private_key),)
+
+    async with bob_alexandria_client.subscribe(LocateMessage) as subscription:
+
+        async def _respond():
+            request = await subscription.receive()
+            await bob_alexandria_client.send_locations(
+                request.sender_node_id,
+                request.sender_endpoint,
+                advertisements=advertisements,
+                request_id=request.request_id,
+            )
+
+        async with trio.open_nursery() as nursery:
+            nursery.start_soon(_respond)
+
+            responses = await alice_alexandria_client.locate(
+                bob.node_id,
+                bob.endpoint,
+                content_key=advertisements[0].content_key,
+                request_id=b"\x01",
+            )
+            assert len(responses) == 1
+            response = responses[0]
+            assert isinstance(response.message, LocationsMessage)
+            assert response.message.payload.total == 1
+            assert response.message.payload.locations == advertisements
```

Comment:
nitpick, but it looks like the returned `request_id` isn't being tested in this and the following test.